cilium/cmd/bpf_metrics_list.go

```diff
@@ -129,6 +165,44 @@ func listMetrics(bpfMetricsList map[string][]string) {
 	w.Flush()
 }
 
+func extractRow(key, value string) (*metricsRow, error) {
+	reasonCodeStr, directionCodeStr, ok := extractTwoValues(key)
+	if !ok {
+		return nil, fmt.Errorf("Cannot extract reason and traffic direction from map's key \"%s\"", key)
+	}
+
+	reasonCode, err := strconv.Atoi(reasonCodeStr)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse reason: %s", err)
+	}
+
+	reasonDesc := monitorAPI.DropReason(uint8(reasonCode))
+
+	directionCode, err := strconv.Atoi(directionCodeStr)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse direction: %s", err)
+	}
+
+	direction := metricsmap.MetricDirection(uint8(directionCode))
+
+	packetsStr, bytesStr, ok := extractTwoValues(value)
+	if !ok {
+		return nil, fmt.Errorf("Cannot extract packets and bytes counters from map's value \"%s\"", value)
+	}
+
+	packets, err := strconv.Atoi(packetsStr)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse packets counter: %s", err)
+	}
+
+	bytes, err := strconv.Atoi(bytesStr)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot parse bytes counter: %s", err)
+	}
+
+	return &metricsRow{reasonCode, reasonDesc, direction, packets, bytes}, nil
```

Comment:
Style nit, feel free to ignore. I just reordered the statements that don't return an error together so that the pattern of the code is consistent with functions that return error, to functions that don't return error.

```suggestion
	directionCode, err := strconv.Atoi(directionCodeStr)
	if err != nil {
		return nil, fmt.Errorf("Cannot parse direction: %s", err)
	}

	packetsStr, bytesStr, ok := extractTwoValues(value)
	if !ok {
		return nil, fmt.Errorf("Cannot extract packets and bytes counters from map's value \"%s\"", value)
	}

	packets, err := strconv.Atoi(packetsStr)
	if err != nil {
		return nil, fmt.Errorf("Cannot parse packets counter: %s", err)
	}

	bytes, err := strconv.Atoi(bytesStr)
	if err != nil {
		return nil, fmt.Errorf("Cannot parse bytes counter: %s", err)
	}
	
	reasonDesc := monitorAPI.DropReason(uint8(reasonCode))
	direction := metricsmap.MetricDirection(uint8(directionCode))

	return &metricsRow{reasonCode, reasonDesc, direction, packets, bytes}, nil
```